### PR TITLE
Bugfix/netdata crashing if not available

### PR DESCRIPTION
**Pull request by vicervin**

### CODE REVIEW

File: s64da_benchmark_toolkit/netdata.py [hunk 1]
```diff
@@ -8,6 +8,21 @@
 
 LOG = logging.getLogger()
 
+def is_netdata_set_and_runnning(config):
+    netdata_config = config.get('netdata')
+    if netdata_config:
+        try:
+            requests.get(netdata_config['url'])
+            return True
```
Comment (sdressler) [inline s64da_benchmark_toolkit/netdata.py:16]: That's not enough, IMO. You want to check for the status code being 200.

File: s64da_benchmark_toolkit/streams.py [hunk 2]
```diff
@@ -82,6 +83,8 @@ def apply_sql_modifications(sql, modifications):
         return sql
 
     def run(self):
+        if not is_netdata_set_and_runnning(self.config):
+            return
```
Comment (sdressler) [inline s64da_benchmark_toolkit/streams.py:87]: Is this intentional?

Comment (oferz) [reply s64da_benchmark_toolkit/streams.py:87]: Yes. The thinking behind it was that if you're running the toolkit and expecting to also get netdata data collected, you'll want to know if this is not going to happen ASAP and not only at the end of the run.
However, this was assuming that you explicitly asking the tool to collect netdata data. If netdata data is collected by default, then this is not required here.

File: s64da_benchmark_toolkit/netdata.py [hunk 3]
```diff
@@ -8,6 +8,21 @@
 
 LOG = logging.getLogger()
 
+def is_netdata_set_and_runnning(config):
+    netdata_config = config.get('netdata')
```
Comment (sdressler) [inline s64da_benchmark_toolkit/netdata.py:12]: Maybe even shorten to:

    netdata_url = config.get('netdata').get('url')
    if netdata_url:
        ...

File: s64da_benchmark_toolkit/netdata.py [hunk 4]
```diff
@@ -8,6 +8,25 @@
...
+def is_netdata_set_and_runnning(config):
+    netdata_url = config.get('netdata').get('url')
+    if netdata_url:
+        try:
+            response = requests.get(netdata_url)
+            status_code = response.status_code
+            if status_code != 200:
+                LOG.error(f'Netdata url response ({netdata_url}) does not return 200, but {status_code}')
+                return False
+            return True
```
Comment (oferz) [inline s64da_benchmark_toolkit/netdata.py:20]: This `return True` can be removed, along with the `else` on line 27 to simplify the logic.

File: s64da_benchmark_toolkit/reporting.py [hunk 5]
```diff
@@ -96,7 +96,7 @@ def run_report(self, reporting_queue):
         netdata_config = self.config.get('netdata')
         if netdata_config:
             netdata = Netdata(netdata_config)
-            if self.netdata_output_file:
+            if self.netdata_output_file and netdata.is_netdata_set_and_running:
```
Comment (oferz) [inline s64da_benchmark_toolkit/reporting.py:99]: This should be:
`and self.is_netdata_set_and_running():`

File: s64da_benchmark_toolkit/streams.py [hunk 6]
```diff
@@ -17,6 +17,7 @@
 
 from .db import DB
 from .reporting import Reporting, QueryMetric
+from .netdata import is_netdata_set_and_runnning
```
Comment (oferz) [inline s64da_benchmark_toolkit/streams.py:20]: Looks like this can be removed now.

File: s64da_benchmark_toolkit/reporting.py (hunk 5 above)
Comment (sdressler) [reply s64da_benchmark_toolkit/reporting.py:99]: Also, IMO it'd be better to move `netdata_output_file` into `Netdata`, then you have it encapsulated.

**Pull request merged**